pkg/kubelet/cri/remote/utils_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func TestVerifySandboxStatus(t *testing.T) {
+	validStatus := getValidPodSandboxStatus()
+	statusWithoutID := getValidPodSandboxStatus()
+	statusWithoutID.Id = ""
+	statusWithoutMetadata := getValidPodSandboxStatus()
+	statusWithoutMetadata.Metadata = nil
+	statusWithoutCreatedAt := getValidPodSandboxStatus()
+	statusWithoutCreatedAt.CreatedAt = 0
+	statusWithoutName := getValidPodSandboxStatus()
+	statusWithoutName.Metadata.Name = ""
+
+	testcases := []struct {
+		description      string
+		podSandboxStatus *runtimeapi.PodSandboxStatus
+		expectedResult   error
+	}{{
+		description:      "verifySandboxStatus succeeds",
+		podSandboxStatus: validStatus,
+		expectedResult:   nil,
+	}, {
+		description:      "verifySandboxStatus fails due to lack of Id",
+		podSandboxStatus: statusWithoutID,
+		expectedResult:   fmt.Errorf("status.Id is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of metadata",
+		podSandboxStatus: statusWithoutMetadata,
+		expectedResult:   fmt.Errorf("status.Metadata is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of CreatedAt",
+		podSandboxStatus: statusWithoutCreatedAt,
+		expectedResult:   fmt.Errorf("status.CreatedAt is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of name",
+		podSandboxStatus: statusWithoutName,
+		expectedResult:   fmt.Errorf("metadata.Name, metadata.Namespace or metadata.Uid is not in metadata \"&PodSandboxMetadata{Name:,Uid:100,Namespace:foo-ns,Attempt:10,}\""),
+	}}
+
+	for _, tc := range testcases {
+		t.Run(tc.description, func(t *testing.T) {
+			result := verifySandboxStatus(tc.podSandboxStatus)
```

Comment:
If we're adding unit test coverage for one this file, can we also add some for `verifyContainerStatus`? https://github.com/kubernetes/kubernetes/blob/06a7e2bacfa4f15d3fdfe743f9521125e80cf754/pkg/kubelet/cri/remote/utils.go#L64 Should be pretty straightforward.

Reply:
Do you want to see more unit tests for the other function verifyContainerStatus() in this PR?
Or is it fine to add these unit tests in another PR?

My concern is more unit tests make this PR bigger and it would be hard to be reviewed.

Reply:
I think you should do it all as part of one PR, since they are for the same file.

Reply:
OK, then can you provide what I should do on the above comment?
because the unit tests of the other function also tend to be similar.

Reply:
Yes, please add unit tests for `verifyContainerStatus`, similar to what you've done for `verifySandboxStatus`.

Reply:
OK, thanks for your explanation.
I will add new unit tests based on current implementation.

Reply:
done on the latest PR.